### PR TITLE
add:all_redus_files

### PR DESCRIPTION
**Pull request by artemiusV**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,15 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Home />} />
```
Comment (AdammmVV) [inline src/App.tsx]: create enum for paths

File: src/api.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const { data } = await instanceAxios.get("/search/users?q=" + query, {
+      params: {
+        page,
+        per_page: limit,
+      },
+    });
+    return data?.items;
+  } catch (error) {
+    return null;
+  }
```
Comment (AdammmVV) [inline src/api.tsx:23]: handle errors in thunk

File: src/components/pages/Home/Home.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+          users.map((user) => {
+            return <User user={user} key={user.id} />;
+          })
+        ) : (
+          <h2>There is nothing to display...</h2>
+        )}
+      </div>
+    </div>
+  );
+}; */
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]: delete unused comments here and throughout app

File: src/components/pages/Home/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  const { register, handleSubmit } = useForm();
+  const [users, setUsers] = useState<UsersProps[]>([]);
+  const [page, setPage] = useState<number>(1);
+  const [limit, setLimit] = useState<number>(10);
+
+  const handlePrevPage = () => {
+    setPage((page: number) => {
+      if (page === 1) return page;
+      else return page - 1;
+    });
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]:
```suggestion
    setPage((page: number) => page === 1 ? page : page - 1);
```

File: src/components/pages/Home/Home.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+import "./Home.css";
+import { User } from "../../ui/User";
+import { fetchUsers } from "../../../api";
+
+interface HomeProps {}
+interface UsersProps {
+  avatar_url: string;
+  login: string;
+  id: number;
+}
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]: put it in separate file typs.ts

File: src/components/pages/Home/Home.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  );
+}; */
+
+import React, { useState } from "react";
+import { SubmitHandler, useForm, FieldValues } from "react-hook-form";
+import "./Home.css";
+import { User } from "../../ui/User";
+import { fetchUsers } from "../../../api";
+
+interface HomeProps {}
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]: what is this for?

File: src/components/pages/Home/Home.tsx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+interface UsersProps {
+  avatar_url: string;
+  login: string;
+  id: number;
+}
+// interface FieldValues {
+//   query: string;
+// }
+
+export const Home: React.FC<HomeProps> = () => {
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]: use: 
```suggestion
export const Home: FC<HomeProps> = () => {
```

File: src/components/pages/Home/Home.tsx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+  const { register, handleSubmit } = useForm();
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx:11]: add types for useForm

File: src/components/pages/Home/Home.tsx [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+  };
+
+  const onSubmit: SubmitHandler<FieldValues> = async (data) => {
+    const query = data.query;
+    if (query) {
+      const items = await fetchUsers(query, page, limit);
+      setUsers(items);
+    } else {
+      console.log("Your query is empty ...");
+    }
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]: put it in Thunk

File: src/components/pages/Home/Home.tsx [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+
+  const handlePrevPage = () => {
+    setPage((page: number) => {
+      if (page === 1) return page;
+      else return page - 1;
+    });
+  };
+
+  const handleNextPage = () => {
+    setPage((page) => page + 1);
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx:25]: If there are no more pages, where will we go?

File: src/components/pages/Home/Home.tsx [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+  const onSubmit: SubmitHandler<FieldValues> = async (data) => {
+    const query = data.query;
+    if (query) {
+      const items = await fetchUsers(query, page, limit);
+      setUsers(items);
+    } else {
+      console.log("Your query is empty ...");
+    }
+  };
```
Comment (AdammmVV) [inline src/components/pages/Home/Home.tsx]: try doing it via Debounce

File: src/components/pages/User/UserProfile.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+  blog: string | null;
+  html_url: string;
+}
+
+export interface RepoInfo {
+  id: number;
+  name: string;
+  description: string;
+  language: string;
+}
```
Comment (AdammmVV) [inline src/components/pages/User/UserProfile.tsx]: put it in types.ts

File: src/components/pages/User/UserProfile.tsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+export const User: React.FC = () => {
```
Comment (AdammmVV) [inline src/components/pages/User/UserProfile.tsx]:
```suggestion
export const User: FC = () => {
```

File: src/components/pages/User/UserProfile.tsx [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+              <a href={userInfo?.html_url}>View GitHub Profile</a>
+            </p>
+          </div>
+        </div>
+      </div>
+      <div className="user-repos">
+        {repos?.length ? (
+          repos.map((repo) => {
+            return <Repo repo={repo} key={repo.id} />;
+          })
```
Comment (AdammmVV) [inline src/components/pages/User/UserProfile.tsx]:
```suggestion
          repos.map((repo) => <Repo repo={repo} key={repo.id} />)
```

File: src/components/ui/Repo.tsx [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+type RepoProps = {
+  repo: {
+    name: string;
+    description: string;
+    language: string | null;
+  };
+};
+
+export const Repo: React.FC<RepoProps> = ({ repo }) => {
```
Comment (AdammmVV) [inline src/components/ui/Repo.tsx]:
```suggestion
export const Repo: FC<RepoProps> = ({ repo }) => {
```

File: src/components/ui/User.tsx [hunk 16]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+interface UserProps {
+  user: {
+    avatar_url: string;
+    login: string;
+    id: number;
+  };
+}
```
Comment (AdammmVV) [inline src/components/ui/User.tsx]: use type for Props
```suggestion
type UserProps = {
  user: {
    avatar_url: string;
    login: string;
    id: number;
  };
}
```

File: src/store/userSlice/userSlice.ts [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+    setUser: (state, action) => { 
+      state.user = action.payload.user
+    },
+    setRepos: (state,action:PayloadAction<{userRepos: RepoInfo[]}>)=>{
+      state.userRepos=action.payload.userRepos
+    },
+  }, 
+   
+})
+
```
Comment (AdammmVV) [inline src/store/userSlice/userSlice.ts:26]: remove empty line

File: src/store/userSlice/userSlice.ts [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+  string,
+  {
+    dispatch: AppDispatch,
+    state: RootState
+    }
+>('user/fetchUserInformation', async (login, thunkApi) => {
+  const respon1 = await fetchUserData(login);
+  const respons2 = await fetchUserRepos(login);
+  thunkApi.dispatch(setUser({user: respon1.data}))
+  thunkApi.dispatch(setRepos({userRepos: respons2.data}))
```
Comment (AdammmVV) [inline src/store/userSlice/userSlice.ts]: create error handling

Comment (artemiusV) [reply src/store/userSlice/userSlice.ts]: add error hending

File: src/store/userSlice/userSlice.ts (hunk 17 above)
Comment (artemiusV) [reply src/store/userSlice/userSlice.ts:26]: remove empty line

File: src/components/ui/User.tsx (hunk 16 above)
Comment (artemiusV) [reply src/components/ui/User.tsx]: fixed the interface to types

File: src/api.tsx (hunk 2 above)
Comment (artemiusV) [reply src/api.tsx:23]: add handle errors

File: src/App.tsx (hunk 1 above)
Comment (artemiusV) [reply src/App.tsx]: create enum

File: src/components/pages/Home/Home.tsx (hunk 6 above)
Comment (artemiusV) [reply src/components/pages/Home/Home.tsx]: suddenly in the future you will need to add props to Home

File: src/components/pages/Home/Home.tsx (hunk 7 above)
Comment (artemiusV) [reply src/components/pages/Home/Home.tsx]: if I use your option I get an error TS2304: Cannot find name 'FC'.

File: src/components/pages/Home/Home.tsx (hunk 4 above)
Comment (artemiusV) [reply src/components/pages/Home/Home.tsx]: formatted code

File: src/components/pages/User/UserProfile.tsx (hunk 14 above)
Comment (artemiusV) [reply src/components/pages/User/UserProfile.tsx]: formatted code

File: src/components/pages/Home/Home.tsx (hunk 7 above)
Comment (artemiusV) [reply src/components/pages/Home/Home.tsx]: corrected the mistake

File: src/components/pages/User/UserProfile.tsx (hunk 13 above)
Comment (artemiusV) [reply src/components/pages/User/UserProfile.tsx]: create mistake

File: src/components/pages/User/UserProfile.tsx (hunk 12 above)
Comment (artemiusV) [reply src/components/pages/User/UserProfile.tsx]: add type in types

File: src/components/pages/Home/Home.tsx (hunk 5 above)
Comment (artemiusV) [reply src/components/pages/Home/Home.tsx]: add type in types

File: src/components/ui/Repo.tsx (hunk 15 above)
Comment (artemiusV) [reply src/components/ui/Repo.tsx]: formated code

File: src/components/pages/Home/Home.tsx (hunk 10 above)
Comment (artemiusV) [reply src/components/pages/Home/Home.tsx:25]: I added a check
Don't enlarge page if there is no data

**Pull request merged**